inc/saitypes.h

```diff
@@ -1003,6 +1003,127 @@ typedef struct _sai_port_eye_values_list_t
     sai_port_lane_eye_values_t *list;
 } sai_port_eye_values_list_t;
 
+/**
+ * @brief Defines a lane with its list of CTLE values.
+ *
+ * Caller must allocate the buffer for the list member and set the count
+ * member to the size of the allocated objects in the list member.
+ *
+ * If the size is large enough to accommodate the list of objects, the
+ * callee must fill the list member and set the count member to the actual
+ * number of objects filled. If the size is not large enough, the callee
+ * must set the count member to the actual number of objects filled in the
+ * list member and return #SAI_STATUS_BUFFER_OVERFLOW. Once the caller
+ * gets such a return code, it may use the returned count member to
+ * re-allocate the list and retry.
+ */
+typedef struct _sai_port_lane_ctle_values_t
+{
+    uint32_t lane;
+    sai_u32_list_t ctle;
+
+} sai_port_lane_ctle_values_t;
+
+/**
+ * @brief Defines a port's lanes CTLE values list
+ *
+ * In get_port_attribute function call, the count member defines the number
+ * of objects which will be returned to the caller in the list member. The
+ * caller must allocate the buffer for the list member and set the count
+ * member to the size of the allocated objects in the list member.
+ *
+ * If the size is large enough to accommodate the list of objects, the
+ * callee must fill the list member and set the count member to the actual
+ * number of objects filled. If the size is not large enough, the callee
+ * must set the count member to the actual number of objects filled in the
+ * list member and return #SAI_STATUS_BUFFER_OVERFLOW. Once the caller
+ * gets such a return code, it may use the returned count member to
+ * re-allocate the list and retry.
+ */
+typedef struct _sai_port_ctle_values_list_t
+{
+    uint32_t count;
+    sai_port_lane_ctle_values_t *list;
```

Comment:
this is like a list in list, so its like a 2 dimensional array, is that expected?

Reply:
yes, that's expected,  since these attributes return the per lane values for port, and CTLE and FFE coefficients return as a list. Accordingly, as result, we have the list of lanes and the list of values for every lane.